src/pdesc.c

```diff
@@ -111,6 +111,13 @@ proxy_desc_t *create_and_insert_proxy_desc(uint16_t id_no, uint16_t icmp_id,
 	pthread_mutex_unlock(&chain_lock);
 	cur->xfer.bytes_in      = 0.0;
 	cur->xfer.bytes_out     = 0.0;
+	cur->window_size	= opts.window_size ? opts.window_size : 64;
```

Comment:
We might need some validation I guess. What happens if we use a windows_size of 1?

Reply:
Nothing happens. Window size 1 is valid, but the connection hangs when the window immediately becomes full.

The minimum viable window size to transfer data packets is window size 2. However a practical window size should be large enough to hold the number of explicit acknowledgements which can be lost during one minute, otherwise the window becomes full of lost acknowledgements. Not by coincidence, the default window size is 64 which is larger than the number of explicit acknowledgements in 60 seconds at the default interval of 1000 milliseconds.